### DISCUSSION TITLE
Theia launched from TheiaBlueprint hangs on Win10

### DISCUSSION THREAD

**Discussion by Springvar** [eclipse-theia/theia#11052] [Q&A] [answered]
I'm trying to use TheiaBlueprint (and another self-made Theia based IDE) to build and launch Theia (and this other Theia based IDE), but it ends up getting stuck on the loading indicator.

Using the terminal, I enter the commands
> yarn
> yarn build
> yarn electron start:debug

This works fine when running from the native cmd terminal, or through vscode, but not from the Theia terminal. I've tried to replace the integrated shell to powershell.exe, but the problem remains the same.

Is this a known issue, and are there any known solutions?

**Comment by vince-fugnitto** [comment, marked as answer]
@Springvar thank you for the discussion, I don't believe it is a known issue, I never experience the problem.
Do you have any additional details that may help? When stuck on the loading screen what do you see in the frontend (devtools) and backend logs?

**Reply by Springvar** [reply]
The backend log ends with
`root INFO Deploy plugins list: 24754.0 ms [Finished 26.382 s after backend start]`
with no indication that anything has gone wrong before that.

DevTools inspection of frontend indicates that it is not able to establish a ws connection to the backend

![image](https://user-images.githubusercontent.com/3775536/164491863-7df11e4a-2300-48e4-98bf-1926a7b8c4ce.png)

**Reply by jfaltermeier** [reply]
I've started to look into the issue and can give a small update. It is also reproducible without Blueprint and with using the Theia electron example application. 
The ws upgrade is not allowed because the `ElectronTokenValidator#isTokenValid` returns false. 

Values from Theia Electron example application launched from VSCode:
```
received f27d5652-2024-42c6-ae0d-c6e18317ca6f
expected f27d5652-2024-42c6-ae0d-c6e18317ca6f
```

Values from second Theia Electron example application launched from the first one:
```
received 1fd68d02-e6ef-48d8-b2d4-f8be27cb6424
expected f27d5652-2024-42c6-ae0d-c6e18317ca6f
```

So the nested electron app for some reason expects the same token as for its parent.

**Reply by jfaltermeier** [reply]
https://nodejs.org/api/child_process.html mentions this:
```
On Windows, environment variables are case-insensitive. Node.js lexicographically sorts the env keys and uses the first one that case-insensitively matches. Only first (in lexicographic order) entry will be passed to the subprocess. This might lead to issues on Windows when passing objects to the env option that have multiple variants of the same key, such as PATH and Path.
```

I think this is the issue here. The environment variable from the parent process wins, because it is all upper-cased. I will open an issue with a proposed fix.

**Reply by jfaltermeier** [reply]
The ticket is: https://github.com/eclipse-theia/theia/issues/11081
Fix should be released with 1.25.0: https://github.com/eclipse-theia/theia/milestone/35?closed=1

**Reply by Springvar** [reply]
Great, that worked nicely. Thank you!
